toqito/nonlocal_games/nonlocal_game.py

```diff
@@ -153,20 +176,22 @@ def classical_value(self) -> float:
             ) = pred_mat_copy.shape
         pred_mat_copy = np.transpose(pred_mat_copy, (0, 2, 1, 3))
 
-        for i in range(num_alice_outputs**num_bob_inputs):
-            number = i
-            base = num_bob_outputs
-            digits = num_bob_inputs
-            b_ind = np.zeros(digits)
-            for j in range(digits):
-                b_ind[digits - j - 1] = np.mod(number, base)
-                number = np.floor(number / base)
-            pred_alice = np.zeros((num_alice_outputs, num_alice_inputs))
+        num_iterations = num_alice_outputs**num_bob_inputs
+
+        if num_iterations > 1000:
+            with multiprocessing.Pool() as pool:
```

Comment:
Do we want to maybe consider something like this? 

```py
# Determine tgvals based on iteration count
if num_iterations > 1000:
    with multiprocessing.Pool() as pool:
        tgvals = pool.starmap(
            NonlocalGame.process_iteration,
            [(i, num_bob_outputs, num_bob_inputs, pred_mat_copy, num_alice_outputs, num_alice_inputs)
             for i in range(num_iterations)]
        )
else:
    tgvals = [
        NonlocalGame.process_iteration(i, num_bob_outputs, num_bob_inputs, pred_mat_copy, num_alice_outputs, num_alice_inputs)
        for i in range(num_iterations)
    ]

# Compute maximum win probability
p_win = max(tgvals)
```

Reply:
Yes, we can do this, but I think it doesn't affect the complexity. But if you want I can modify it to this.

Reply:
Yep. Since the tgval calculation is essentially the same in parallel and serial cases, you could refactor it into a single calculation to reduce redundancy. For instance, store the results in a list, then use max(tgvals) afterward. The way you do it is fine as well, so I'm okay with either approach!